### DISCUSSION TITLE
(Help wanted) Change of the Butcher Quality system.

### DISCUSSION THREAD

**Discussion by Lamandus** [cataclysmbn/Cataclysm-BN#2054] [General]
# Describe the Problem
Right now, the butcher quality is using a different system like we know from other tools. Since the yield is not affected by the butcher quality, it would make sense to cut it down to ~5 butcher qualities.

# How to solve
To quote coolchulu:  1 - unsuited, like a sharp rock (2x-4x time), 2 - OK, a large knife (normal), 3 - specialized bulky item (0.75x time), 4 - ???, 5 - power tool, like chainsaw (0.5x time).
The part which seem to affect it, is found in activity_handlers.cpp . Time_to_cut is the amount of time needed to butcher.
So if anyone is willing to do it, feel free to start out. I am a bit on the short end of time. That's why I post it here.

**Comment by chaosvolt** [comment]
Translating the existing ranges might work as a start:
* Anything negative: Level 1 (covers sharp rocks, flailing around with a chainsaw latajang or other super-awkward tool, etc)
* 1-10: Level 2 (covers makeshift knives, most swords that aren't made for use as a tool, etc).
* 11-20: Level 3 (covers basic level knives, more wieldy swords).
* 21-30: Level 4 (covers everything up to the hunting knife, which has the highest butchering quality of any non-scifi item).
* 30+: Level 5 (scifi stuff, currently only autonomous surgical scalpel counts).

Qualities could then be adjusted as desired, i.e. if we want chainsaws to count as a super-fast butchery item we'd move it from level 1 (as it currently has -100 butchering quality) to level 5.

**Comment by Lamandus** [comment]
> Qualities could then be adjusted as desired, i.e. if we want chainsaws to count as a super-fast butchery item we'd move it from level 1 (as it currently has -100 butchering quality) to level 5.

but only the switched on, one.

**Comment by Coolthulhu** [comment]
We'd need a new name. `MEAT_CUT`? `CUT_FLESH`?
At first, butchery could be hardcode-translated into the new quality using something like CV outlined.
Mods can technically require butchering quality for crafts. That's in requirements class, so just one place to cover all crafting-related changes. I'd prefer a warning instead of translating to new quality, because we don't have examples of butcher quality used for crafting.

> but only the switched on, one.

Ideally, a wielded one that is turned off would print an extra message and consume some charges to cut faster, since 0.5 time would most likely end up costing a ton of chainsaw fuel. But it's separate from this issue.

**Comment by Relgar** [comment]
It may offer an additional avenue for rarity-based meat / corpse products, leading to 'advancement' in that sense for a character who is a butcher. For example:

Corpses drop x amounts of chunk of meat. This could be sized up so that it drops fewer of these, but they are larger. From these chunks of meat you could the carve of different parts depending on the quality of the blade available as a recipe.

Quality 1: large pieces of meat.
Quality 2: above + small pieces of mea.
Quality 3: above + specific parts of the meat (leg, flank, rib)
Quality 4: above + nice pieces of meat (scotch fillet, rump, porterhouse)
Quality 5: above + scifi level of meat heaven (cube of beef essence or what have you)
